Testing issue: Unit test for `PhasorDynamics/Load` model Jacobian fails

### Failed test outputs or steps to reproduce the failure(s)

<!-- Include output of `ctest -VV --rerun-failed`. -->        
```console
$ ctest -VV --rerun-failed
Test project /Users/55y/src/gridkit/build-gcc
    Start 23: PhasorDynamicsLoadTest
1/1 Test #23: PhasorDynamicsLoadTest ...........***Failed    0.12 sec
--- PASS: Test constructor
--- PASS: Test residual
[Utils/Testing.hpp]: Mismatching map values! a.first = 1, a.second = -0.2, and b.second = 0.2
[Utils/Testing.hpp]: Mismatching map values! a.first = 0, a.second = 0.2, and b.second = -0.2
--- FAIL: Test jacobian


Test Summary
----------------------------
        Successful tests:     2
        Failed test:          1
        Skipped tests:        0
        Expected failures:    0
        Unexpected successes: 0



0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.14 sec

The following tests FAILED:
         23 - PhasorDynamicsLoadTest (Failed)
Errors while running CTest
```


### GridKit™ version

`develop` at 9a87777814fec8d2bb9df1375d3e96fadc71e0d8

### System and environment details

Built on MacOS 15.4.1 with GCC14.2.0.

### Additional information

The branch was not rebased to `develop` before merging so it didn't pick up fix to the `PhasorDynamics/Load` model.

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -136,7 +136,6 @@ jobs:
                 padded_length: 128
             mirrors:
               local-buildcache: oci://${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
-              spack: https://binaries.spack.io/v0.23.1
             packages:
               llvm:
                 externals:

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -134,20 +134,12 @@ jobs:
               install_tree:
                 root: /opt/spack
                 padded_length: 128
-            mirrors:
-              local-buildcache: oci://${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
             packages:
               llvm:
                 externals:
                 - spec: llvm@${{ matrix.llvm }}
                   prefix: /usr/lib/llvm-${{ matrix.llvm }}
           EOF
-
-      - name: Configure GHCR mirror
-        run: spack -e . mirror set --oci-username ${{ env.USERNAME }} --oci-password "${{ secrets.GITHUB_TOKEN }}" local-buildcache
-
-      - name: Trust keys
-        run: spack -e . buildcache keys --install --trust
 
       - name: Find external packages
         run: spack -e . external find --all --exclude llvm
@@ -172,9 +164,3 @@ jobs:
 
       - name: Test Build
         run: cd $(spack -e . location --build-dir gridkit@develop) && ctest -VV
-
-      # Push with force to override existing binaries...
-      - name: Push to binaries to buildcache
-        run: |
-          spack -e . buildcache push --force --base-image ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.BASE_VERSION }} --unsigned --update-index local-buildcache
-        if: ${{ !cancelled() }}

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -92,7 +92,7 @@ jobs:
     permissions:
       packages: write
       contents: read
-    timeout-minutes: 30
+    timeout-minutes: 60
 
     strategy:
       matrix:
@@ -141,7 +141,7 @@ jobs:
           EOF
 
       - name: Configure GHCR mirror
-        run: spack -e . mirror add --oci-username ${{ env.USERNAME }} --oci-password "${{ secrets.GITHUB_TOKEN }}" local-buildcache oci://${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+        run: spack -e . mirror add --oci-username ${{ env.USERNAME }} --oci-password "${{ secrets.GITHUB_TOKEN }}" local-buildcache oci://${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}/${{ env.BASE_VERSION }}
 
       - name: Trust keys
         run: spack -e . buildcache keys --install --trust

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -101,7 +101,6 @@ jobs:
         spack_spec:
           - gridkit@develop +enzyme
             ^enzyme@0.0.173
-            ^gcc-runtime@13.3.0
 
     steps:
       - name: Add LLVM
@@ -141,14 +140,17 @@ jobs:
                   prefix: /usr/lib/llvm-${{ matrix.llvm }}
           EOF
 
+      - name: Configure GHCR mirror
+        run: spack -e . mirror add --oci-username ${{ env.USERNAME }} --oci-password "${{ secrets.GITHUB_TOKEN }}" local-buildcache oci://${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Trust keys
+        run: spack -e . buildcache keys --install --trust
+
       - name: Find external packages
         run: spack -e . external find --all --exclude llvm
 
       - name: Spack develop gridkit
         run: spack -e . develop --path=$(pwd) gridkit@develop
-
-      - name: Spack clean
-        run: spack clean -am
 
       - name: List working directory
         run: ls -ltrh $(pwd)/*/*/
@@ -164,3 +166,9 @@ jobs:
 
       - name: Test Build
         run: cd $(spack -e . location --build-dir gridkit@develop) && ctest -VV
+
+      # Push with force to override existing binaries...
+      - name: Push to binaries to buildcache
+        run: |
+          spack -e . buildcache push --force --base-image ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.BASE_VERSION }} --unsigned --update-index local-buildcache
+        if: ${{ !cancelled() }}

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -133,6 +133,9 @@ jobs:
               install_tree:
                 root: /opt/spack
                 padded_length: 128
+            mirrors:
+              local-buildcache: oci://${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}/${{ env.BASE_VERSION }}
+              spack: https://binaries.spack.io/v0.23.1
             packages:
               llvm:
                 externals:
@@ -141,7 +144,7 @@ jobs:
           EOF
 
       - name: Configure GHCR mirror
-        run: spack -e . mirror add --oci-username ${{ env.USERNAME }} --oci-password "${{ secrets.GITHUB_TOKEN }}" local-buildcache oci://${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}/${{ env.BASE_VERSION }}
+        run: spack -e . mirror set --oci-username ${{ env.USERNAME }} --oci-password "${{ secrets.GITHUB_TOKEN }}" local-buildcache
 
       - name: Trust keys
         run: spack -e . buildcache keys --install --trust
@@ -159,7 +162,7 @@ jobs:
         run: spack -e . concretize
 
       - name: Install dependencies
-        run: spack -e . install --no-check-signature --no-cache --fresh --only dependencies
+        run: spack -e . install --no-check-signature --only dependencies
 
       - name: Install package
         run: spack -e . install --keep-stage --no-check-signature --no-cache --fresh

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -101,6 +101,7 @@ jobs:
         spack_spec:
           - gridkit@develop +enzyme
             ^enzyme@0.0.173
+            ^gcc-runtime@13.3.0
 
     steps:
       - name: Add LLVM

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -155,6 +155,9 @@ jobs:
       - name: Spack develop gridkit
         run: spack -e . develop --path=$(pwd) gridkit@develop
 
+      - name: Spack clean
+        run: spack clean -am
+
       - name: List working directory
         run: ls -ltrh $(pwd)/*/*/
 

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -161,7 +161,7 @@ jobs:
         run: spack -e . concretize
 
       - name: Install dependencies
-        run: spack -e . install --no-check-signature --only dependencies
+        run: spack -e . install --no-check-signature --no-cache --fresh --only dependencies
 
       - name: Install package
         run: spack -e . install --keep-stage --no-check-signature --no-cache --fresh

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -1,5 +1,5 @@
 # https://spack.readthedocs.io/en/latest/binary_caches.html#spack-build-cache-for-github-actions
-name: Spack Ubunutu x86_64 Buildcache
+name: CPU tests (Ubunutu x86_64)
 
 env:
   SPACK_COLOR: always

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -7,7 +7,7 @@ env:
   # Our repo name contains upper case characters, so we can't use ${{ github.repository }}
   IMAGE_NAME: gridkit
   USERNAME: gridkit-bot
-  BASE_VERSION: ubuntu-24.04-fortran-v0.2.11
+  BASE_VERSION: ubuntu-24.04-fortran-v0.2.12
 
 # Until we remove the need to clone submodules to build, this should on be in PRs
 on: [pull_request]

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -92,7 +92,7 @@ jobs:
     permissions:
       packages: write
       contents: read
-    timeout-minutes: 60
+    timeout-minutes: 30
 
     strategy:
       matrix:

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -92,7 +92,7 @@ jobs:
     permissions:
       packages: write
       contents: read
-    timeout-minutes: 30
+    timeout-minutes: 60
 
     strategy:
       matrix:

--- a/.github/workflows/spack_default_build.yaml
+++ b/.github/workflows/spack_default_build.yaml
@@ -128,6 +128,7 @@ jobs:
             specs:
             - ${{ matrix.spack_spec }} target=x86_64_v2
             concretizer:
+              unify: true
               reuse: dependencies
             config:
               install_tree:

--- a/tests/UnitTests/PhasorDynamics/LoadTests.hpp
+++ b/tests/UnitTests/PhasorDynamics/LoadTests.hpp
@@ -108,9 +108,9 @@ namespace GridKit
         const real_type g = R / (R * R + X * X);
 
         real_type dIr_dVr = -g;
-        real_type dIr_dVi = -b;
+        real_type dIr_dVi = b;
 
-        real_type dIi_dVr = b;
+        real_type dIi_dVr = -b;
         real_type dIi_dVi = -g;
 
         std::vector<Sparse::Variable::DependencyMap> dependencies(2);

